smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/middleware/MiddlewareExecutionGenerator.kt

```diff
@@ -114,11 +111,23 @@ class MiddlewareExecutionGenerator(
             writer.write("  .withExpiration(value: expiration)")
         }
 
+        // Add context values for config fields
+        val configProperties = ctx.integrations
+            .flatMap { it.clientConfigurations(ctx) }
+            .flatMap { it.getProperties(ctx) }
+        configProperties.forEach {
+            renderConfigPropertyToContext(it)
+        }
+
         val serviceShape = ctx.service
         httpProtocolCustomizable.renderContextAttributes(ctx, writer, serviceShape, op)
         writer.write("  .build()")
     }
 
+    open fun renderConfigPropertyToContext(configProperty: ConfigProperty) {
+        // No action by default.  Overridden in subclasses
+    }
+
```

Comment:
This extension point may be redundant; we can use existing extension point for printing custom AWS specific context attributes. See [here](https://github.com/awslabs/aws-sdk-swift/blob/b4a7d85261a5315db5f66f5c90dbc9646e13595c/codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/AWSHTTPProtocolCustomizations.kt#L27-L38). The extension point used by previous link is [here](https://github.com/smithy-lang/smithy-swift/blob/fad1c76d242539f1ee5e29817ce720d45b84583c/smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/middleware/MiddlewareExecutionGenerator.kt#L118).